src/connection/bootstrap_group.rs

```diff
@@ -80,15 +82,29 @@ impl BootstrapGroupRef {
     /// is because the bootstrapping was successful (in which case no failure event will be
     /// auto-fired).
     pub fn terminate_group(&self, is_due_to_success: bool) {
-        let mut group = self.group.borrow_mut();
+        let mut terminators = {
+            let mut group = self.group.borrow_mut();
+            if is_due_to_success {
+                group.is_bootstrap_successful_yet = true;
+            }
 
-        if is_due_to_success {
-            group.is_bootstrap_successful_yet = true;
-        }
+            // We use a `mem::replace` here because `self.group` can be mutably borrowed
+            // twice during `BootstrapGroupRef::drop` (it's called when we remove a connection)
+            mem::replace(&mut group.terminators, Default::default())
+        };
 
-        for (_, mut terminator) in group.terminators.drain() {
-            let _ = terminator.try_send(());
-        }
+        ctx_mut(|c| {
+            for (peer_addr, mut terminator) in terminators.drain() {
+                let _ = terminator.try_send(());
+                if peer_addr != self.peer_addr {
```

Comment:
Do we ever add ourselves to the map ? If we did that would be buggy, and even in that case we should promptly remove the connection anyway ?

Reply:
>If we did that would be buggy, and even in that case we should promptly remove the connection anyway ?
Hmm, not sure what you mean exactly. This list of connections is taken directly from `ctx_mut`, so naturally it will contain _all_ connections that've been made, including a succeeding bootstrapped connection (which is `self.peer_addr`). Why would we require to terminate it?